### ISSUE TITLE
mono28

### ISSUE THREAD

**Issue by GoogleCodeExporter** [noahocker/winetricks#221]
```
#winetricks mono28

What is the expected output? What do you see instead?

Download and installation of mono28
Looks like the mirror is down?

What version of the product are you using? On what operating system?
Latest

Please provide any additional information below.

http://ftp.novell.com/pub/mono/archive/2.8.2/windows-installer/1/mono-2.8.2-gtks
harp-2.12.10-win32-1.exe to /home/ruffles/.cache/winetricks/mono28
--2012-06-21 11:39:27--  
http://ftp.novell.com/pub/mono/archive/2.8.2/windows-installer/1/mono-2.8.2-gtks
harp-2.12.10-win32-1.exe
Resolving ftp.novell.com (ftp.novell.com)... 130.57.1.88
Connecting to ftp.novell.com (ftp.novell.com)|130.57.1.88|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2012-06-21 11:39:27 ERROR 404: Not Found.

------------------------------------------------------
Downloading 
http://ftp.novell.com/pub/mono/archive/2.8.2/windows-installer/1/mono-2.8.2-gtks
harp-2.12.10-win32-1.exe failed
------------------------------------------------------
ruffles@ruffles-pc:~$ 

```

Original issue reported on code.google.com by `KevinKad...@gmail.com` on 21 Jun 2012 at 6:41

**Comment by GoogleCodeExporter** [comment]
```
Maybe use this path instead?
http://download.mono-project.com/archive/2.8/windows-installer/9/mono-2.8-gtksha
rp-2.12.10-win32-9.exe
```

Original comment by `ribeiro....@gmail.com` on 26 Jul 2012 at 11:59

**Comment by GoogleCodeExporter** [comment]
```
This happens also with mono 2.10
```

Original comment by `ricardop...@gmail.com` on 22 Oct 2012 at 4:47

**Comment by GoogleCodeExporter** [comment]
```
@ribeiro.breno: That URL is valid, but how do I get winetricks to use that path 
instead of the ftp.novell.com URL? winetricks doesn't ask me for a URL, it just 
magically "knows" to use the one that happens to be broken.

Can somebody post a command line that works to install mono-2.8 (or 2.10, or 
whatever the current version is)?
```

Original comment by `arthur.j...@gmail.com` on 10 Dec 2012 at 7:33

**Comment by GoogleCodeExporter** [comment]
```
The winetricks script has a bunch of hard-coded filenames and URLs in it.  The 
one that came with my distro repo has the broken link, but the newer one at 
http://winetricks.org/winetricks doesn't even support mono28 at all.

Try this patch:

--- /usr/bin/winetricks 2012-03-13 05:34:53.000000000 -0700
+++ ~/Dropbox/Terraria/winetricks-patched-mono28 2013-01-06 00:25:29.877818694 
-0800
@@ -5287,14 +5287,14 @@
     publisher="Novell" \
     year="2010" \
     media="download" \
-    file1="mono-2.8.2-gtksharp-2.12.10-win32-1.exe" \
-    installed_file1="$W_PROGRAMS_X86_WIN/Mono-2.8.2/bin/mono-2.0.dll"
+    file1="mono-2.8-gtksharp-2.12.10-win32-9.exe" \
+    installed_file1="$W_PROGRAMS_X86_WIN/Mono-2.8/bin/mono-2.0.dll"

 load_mono28()
 {
-    # Load Mono, have it handle all .net requests
-    w_download 
http://ftp.novell.com/pub/mono/archive/2.8.2/windows-installer/1/mono-2.8.2-gtks
harp-2.12.10-win32-1.exe d0ee2360b6fb7f16c35b54ee67044ff22bb1487e 
-    w_try $WINE "$W_CACHE"/mono28/mono-2.8.2-gtksharp-2.12.10-win32-1.exe 
$W_UNATTENDED_SLASH_SILENT
+    # Load Mono, have it handle all .net requests -- sha1sum is optional, but 
geeky-cool
+    w_download 
http://download.mono-project.com/archive/2.8/windows-installer/9/mono-2.8-gtksha
rp-2.12.10-win32-9.exe db56dae6ae4eb50a15f40b4a27cdc74654e07ac2
+    w_try $WINE "$W_CACHE"/mono28/mono-2.8-gtksharp-2.12.10-win32-9.exe 
$W_UNATTENDED_SLASH_SILENT
 }

 #----------------------------------------------------------------

```

Original comment by `mar...@gmail.com` on 6 Jan 2013 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
Mono28 is no longer in winetricks (wine ships its own mono).
```

Original comment by `austinenglish@gmail.com` on 29 Jun 2013 at 1:37
- Changed state: **Done**
